### ISSUE TITLE
Ambiguity for annotations in vuln attestation

### ISSUE THREAD

**Issue by lumjjb** [in-toto/attestation#390]
In the [vuln attestation](https://github.com/in-toto/attestation/blob/main/spec/predicates/vuln.md), annotations are specified as an optional list, but the list is not typed. 

Would this be a list of strings `["foo=bar"]`, or a struct
`[ {{"key":"foo"}, {"value": "bar"}}]`, or a list of single maps? `[{"foo": "bar"}]`

```
scanner.result.[*].vulnerability.annotations, optional list

This is a list of key/value pairs where scanners can add additional custom information.
```

**Comment by hectorj2f** [comment]
@lumjjb This would be more of a map <string, value> in my opinion.

**Comment by lumjjb** [comment]
Resolved by #345 through introducing new version 0.2
